src/python/pants/backend/python/typecheck/mypy/rules.py

```diff
@@ -167,15 +170,17 @@ async def mypy_typecheck(
         ),
         python_setup,
     )
-    use_subsystem_constraints = (
-        not mypy.options.is_default("interpreter_constraints")
-        or code_interpreter_constraints.includes_python2()
-    )
-    tool_interpreter_constraints = (
-        PexInterpreterConstraints(mypy.interpreter_constraints)
-        if use_subsystem_constraints
-        else code_interpreter_constraints
-    )
+
+    if not mypy.options.is_default("interpreter_constraints"):
```

Comment:
Is this block mypy-specific? Or could it be factored into a method on `PythonToolBase`?

Reply:
Yes, very MyPy specific. And also going to change super soon to instead be:

```python
tool_interpreter_constraints = (
  ("CPython>=3.8",)
  if mypy.options.is_default("interpreter_constraints") and code_interpreter_constraints.requires_python38_or_newer()
  else mypy.interpreter_constraints
)
```

(This is possible because we're going to set `python_version` automatically, so we no longer care about what interpreter is used to run MyPy. We only care that if there's Py38, we use Py38 because otherwise MyPy can't parse the AST due to `typed-ast` not working with Py38.)